multiaccounts/settings_notifications/database.go

```diff
@@ -0,0 +1,245 @@
+package notificationssettings
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+type SettingType int
+
+const (
+	String SettingType = iota
+	Integer
+	Bool
+)
+
+// Columns' names
+const ColumnID = "id"
+const ColumnExemption = "exemption"
+const ColumnTextValue = "text_value"
+const ColumnIntValue = "int_value"
+const ColumnBoolValue = "bool_value"
+const ColumnExMuteAllMessages = "ex_mute_all_messages"
+const ColumnExPersonalMentions = "ex_personal_mentions"
+const ColumnExGlobalMentions = "ex_global_mentions"
+const ColumnExOtherMessages = "ex_other_messages"
+
+// Static ids we use for notifications settings
+const KeyAllowNotifications = "AllowNotifications"
+const KeyOneToOneChats = "OneToOneChats"
+const KeyGroupChats = "GroupChats"
+const KeyPersonalMentions = "PersonalMentions"
+const KeyGlobalMentions = "GlobalMentions"
+const KeyAllMessages = "AllMessages"
+const KeyContactRequests = "ContactRequests"
+const KeyIdentityVerificationRequests = "IdentityVerificationRequests"
+const KeySoundEnabled = "SoundEnabled"
+const KeyVolume = "Volume"
+const KeyMessagePreview = "MessagePreview"
+
+// Possible values some of ids may have
+const ValueSendAlerts = "SendAlerts"
+const ValueDeliverQuietly = "DeliverQuietly"
+const ValueTurnOff = "TurnOff"
+
+type NotificationsSettings struct {
+	db *sql.DB
+}
+
+func NewNotificationsSettings(db *sql.DB) *NotificationsSettings {
+	return &NotificationsSettings{
+		db: db,
+	}
+}
+
+func (ns *NotificationsSettings) buildSelectQuery(column string) string {
+	return fmt.Sprintf("SELECT %s FROM notifications_settings WHERE id = ?", column)
+}
+
+func (ns *NotificationsSettings) buildInsertOrUpdateQuery(isExemption bool, settingType SettingType) string {
+	var values string
+	if isExemption {
+		values = "VALUES(?, 1, NULL, NULL, NULL, ?, ?, ?, ?)"
+	} else {
+		switch settingType {
+		case String:
+			values = "VALUES(?, 0, ?, NULL, NULL, NULL, NULL, NULL, NULL)"
+		case Integer:
+			values = "VALUES(?, 0, NULL, ?, NULL, NULL, NULL, NULL, NULL)"
+		case Bool:
+			values = "VALUES(?, 0, NULL, NULL, ?, NULL, NULL, NULL, NULL)"
+		}
+	}
+
+	return fmt.Sprintf(`INSERT INTO notifications_settings (
+		id, 
+		exemption, 
+		text_value, 
+		int_value, 
+		bool_value, 
+		ex_mute_all_messages, 
+		ex_personal_mentions,
+		ex_global_mentions,
+		ex_other_messages) 
+		%s;`, values)
+}
+
+// Non exemption settings
+func (ns *NotificationsSettings) GetAllowNotifications() (result bool, err error) {
+	err = ns.db.QueryRow(ns.buildSelectQuery(ColumnBoolValue), KeyAllowNotifications).Scan(&result)
+	return result, err
```

Comment:
One of the errors that these new functions might return is `sql.ErrNoRows`, which happen if the resultset is empty. Should that be considered an error? if not, something like this could be necessary
```go
if err != nil && err != sql.ErrNoRows {
  return false, err
}
```

Reply:
@richard-ramos this depends a lot on the flow you're applying. I added that case you suggested, and returning now default values for each setting in case there is no appropriate row containing that setting in the table, but still in case of other sql errors returning the error, which are properly handled on the desktop app side.